koptional-reactor-extensions/build.gradle

```diff
@@ -7,12 +7,15 @@ dependencies {
 }
 
 dependencies {
-    testCompile libraries.kotlinReflect
-    testCompile libraries.spek
-    testCompile libraries.spekJunitPlatformEngine
-    testCompile libraries.assertJ
-    testCompile libraries.reactorTest
-    testCompile libraries.junitPlatformLauncher
+    testImplementation(libraries.junitPlatformCommons)
+    testImplementation(libraries.junitSpekApi)
+    testImplementation(libraries.kotlinReflect)
+    testImplementation(libraries.reactorTest)
+}
+
+dependencies {
+    testRuntimeOnly(libraries.junitPlatformEngine)
+    testRuntimeOnly(libraries.junitSpekEngine)
```

Comment:
The general idea behind the JUnit Platform — APIs are compile-time, engines are runtime-time (sorry).

Unfortunately Spek does a couple of things wrong:

* the JUnit Platform engine artifact should be a regular Spek dependency but it’s marked as `compileOnly` so we are forced to provide our own;
* the JUnit commons artifact should be a regular Spek dependency to annotate itself with `@Testable` but it doesn’t;
* the Kotlin reflection should not leak outside the engine but it does.